File: 3-Patron/php/CreadorDeMapa.php
<?php
abstract class CreadorDeMapa{
	private $mapa;

	public function __construct($mapa){
    $this->mapa = $mapa;
	}

	public function getMapa(){
		return $this->mapa;	
	}

	public function setMapa($mapa){
		$this->mapa = $mapa;
	}

	abstract public function CrearMapa();
}
?>

File: 3-Patron/Java/CreadorDeMapa.java
package buildercomposite;
public abstract class CreadorDeMapa{
    public IMapa mapa ;
            
    public CreadorDeMapa(IMapa mapa) {
        this.mapa = mapa;
    }
    public IMapa getMapa(){
        return this.mapa;
    }
    public void setMapa(IMapa mapa){
        this.mapa = mapa;
    }
    abstract public Mapa CrearMapa();
}

